### PR TITLE
Users: Use Remote Cache for storing signed in users [v9.3.x] (#59883)

### PR DESCRIPTION
**Pull request by gamab**
Co-authored-by: ievaVasiljeva <ieva.vasiljeva@grafana.com>

Cherry-picked from #59883 

**What is this feature?**

This PR allows using a remote cache to store signed in users (nice to have for HA). 
This work is behind the `"userRemoteCache"` feature toggle.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/user/userimpl/user.go [hunk 1]
```diff
@@ -42,6 +50,13 @@ func ProvideService(
 		cfg:          cfg,
 		teamService:  teamService,
 		cacheService: cacheService,
+		remoteCache:  remoteCache,
+		features:     features,
+		logger:       log.New("user.service"),
```
Comment (gamab) [inline pkg/services/user/userimpl/user.go:55]: Any suggestions for the logger name?

Comment (bergquist) [reply pkg/services/user/userimpl/user.go:55]: This is fine :) :+1:

**Review by bergquist — APPROVED** [review]
LGTM! But having someone else reviewing this as well would be great

**Review by kalleep — APPROVED** [review]
LGTM

**Review by IevaVasiljeva — APPROVED** [review]
Nice! Should we also create a small PR to add logging for user cache to 9.2.x?

File: pkg/services/featuremgmt/registry.go [hunk 2]
```diff
@@ -344,6 +344,10 @@ var (
 		{
 			Name:        "mysqlAnsiQuotes",
 			Description: "Use double quote to escape keyword in Mysql query",
+		},
```
Comment (gamab) [inline pkg/services/featuremgmt/registry.go:347]:
```suggestion
				State: FeatureStateAlpha,
		},
```

**Review by Jguer — APPROVED** [review]

**Pull request merged**